src/renderer/routes/current-path-parameters.injectable.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import { computed } from "mobx";
+import matchingRouteInjectable from "./matching-route.injectable";
+
+const currentPathParametersInjectable = getInjectable({
+  id: "current-path-parameters",
+
+  instantiate: (di) => {
+    const matchingRoute = di.inject(matchingRouteInjectable);
+
+    return computed((): Record<string, string> => {
+      const match = matchingRoute.get();
```

Comment:
nit, I don't think we should have this injectable at all. Because either you are in the case where you already know you have a defined matching route or you know that you are in the case where you don't in which case the `pathParameters` don't matter.

Reply:
There is cases where you don’t want to know which route you are on, but instead you want to do something with path parameters. e.g. all cluster routes after iframe has been removed.

Reply:
I mean the path parameters should be passed into the component by whatever mounts it. That way it is still agnostic and unit testable but we don't have this defensive code.

Reply:
If we kept doing that only the components would have access to path parameters. However, with this you can access path parameters anywhere. Which is requirement when we get rid of iframes.

Reply:
> If we kept doing that only the components would have access to path parameters. However, with this you can access path parameters anywhere.

This should be the winning argument.